plugin-api/src/main/java/org/sonar/api/server/rule/internal/DefaultRule.java

```diff
@@ -101,6 +99,50 @@ public class DefaultRule extends RulesDefinition.Rule {
     this.deprecatedRuleKeys = Collections.unmodifiableSet(new TreeSet<>(newRule.deprecatedRuleKeys()));
     this.ruleDescriptionSections = newRule.getRuleDescriptionSections();
     this.educationPrincipleKeys = Collections.unmodifiableSet(newRule.educationPrincipleKeys());
+
+    this.type = determineType(newRule);
+    this.severity = determineSeverity(newRule);
+    this.defaultImpacts = determineImpacts(newRule);
+  }
+
+  private RuleType determineType(DefaultNewRule newRule) {
+    RuleType type = newRule.type() == null ? RuleTagsToTypeConverter.convert(newRule.tags()) : newRule.type();
+    if (type != null) {
+      return type;
+    }
+
+    if (shouldUseBackmapping(newRule)) {
+      SoftwareQuality softwareQuality = ImpactMapper.getBestImpactForBackMapping(newRule.defaultImpacts()).getKey();
+      return ImpactMapper.convertToRuleType(softwareQuality);
+    }
+    return RuleType.CODE_SMELL;
+  }
+
+  private String determineSeverity(DefaultNewRule newRule) {
+    String severity = newRule.severity();
+    if (severity != null) {
+      return severity;
+    }
+
+    if (shouldUseBackmapping(newRule)) {
+      Severity impactSeverity = ImpactMapper.getBestImpactForBackMapping(newRule.defaultImpacts()).getValue();
```

Comment:
we should be consistent in the naming BackMapping -> backmapping

Reply:
done